x-pack/plugin/ml/src/main/java/org/elasticsearch/xpack/ml/inference/nlp/tokenizers/NlpTokenizer.java

```diff
@@ -207,14 +208,134 @@ public TokenizationResult.Tokens tokenize(
                 );
             }
         }
-        List<DelimitedToken.Encoded> tokens = new ArrayList<>(innerResultSeq1.tokens);
-        tokens.addAll(innerResultSeq2.tokens);
         return createTokensBuilder(clsTokenId(), sepTokenId(), isWithSpecialTokens()).addSequencePair(
             tokenIdsSeq1.stream().map(DelimitedToken.Encoded::getEncoding).collect(Collectors.toList()),
             tokenPositionMapSeq1,
             tokenIdsSeq2.stream().map(DelimitedToken.Encoded::getEncoding).collect(Collectors.toList()),
             tokenPositionMapSeq2
-        ).build(seq1 + seq2, isTruncated, tokens, -1, sequenceId);
+        ).build(List.of(seq1, seq2), isTruncated, List.of(innerResultSeq1.tokens, innerResultSeq2.tokens), -1, sequenceId);
+    }
+
+    /**
+     * Tokenize the two sequences, allowing for spanning of one of the
+     * @param seq1 The first sequence in the pair
+     * @param seq2 The second sequence
+     * @param truncate truncate settings
+     * @param span the spanning settings, how many tokens to overlap.
+     *             We split and span on seq2.
+     * @param sequenceId Unique sequence id for this tokenization
+     * @return tokenization result for the sequence pair
+     */
+    public List<TokenizationResult.Tokens> tokenize(String seq1, String seq2, Tokenization.Truncate truncate, int span, int sequenceId) {
```

Comment:
This looks very similar to 

`tokenize(
        String seq1,
        InnerTokenization innerResultSeq1,
        String seq2,
        Tokenization.Truncate truncate,
        int sequenceId
    )`

but with the spanning logic added. Is there an opportunity to refactor here?

Reply:
Maybe, but the previous one shouldn't do spanning at all. Mainly because its an optimization to allow a single tokenization to be used multiple times. We just have no way of representing a spanned batched encoding (4d tensor?). 

I think some refactoring could be done as a whole, but it would take more churn on the rest of the tokenization code.